### PR TITLE
Progress Controls Template Settings from WUXC + Re-Templating Tests

### LINKED ISSUE #2250: Make ProgressBar and ProgressRing compatible with the WUXC version
The ProgressBar and ProgressRing controls in this repo will be replacing the ones from the framework in WinUI3. However there are a few template settings that are currently missing. This bug is ensure that ProgressBar and ProgressRing are compatible with the ones in the framework including for re-templating scenarios.

### LINKED ISSUE #2266: Allow re-templating ProgressRing with custom lottie animation
It should be possible to re-template the ProgressRing with a custom lottie animation source. It looks like we currently overwrite the source even if one is going to be set in the template.

### LINKED ISSUE #2267: ProgressRing should be collapsed when IsActive is false
Follow the same pattern as WUXC.ProgressRing and hide the control altogether when IsActive is false. This will allow it to be used in a transient way and not have to worry about fiddling with Visibility in the app.

### PR DESCRIPTION
**Pull request by karenbtlai**
## Description

- Adds ProgressBar and ProgressRing Template Settings from WUXC for backwards compatibility
- Check for `!player.Source()` before setting source for re-templating
- Add new test pages for custom lottie animation source and storyboard re-templating with WUXC xaml code
- ProgressRing Visual Update - transition animation time

## Motivation and Context

Fixes #2266 
Fixes #2250 

## How Has This Been Tested?

Existing tests and added tests for custom Lottie source and storyboard re-templating  compatibility with WUXC xaml code.

### CODE REVIEW

File: dev/ProgressRing/ProgressRing.xaml [hunk 1]
```diff
@@ -20,8 +20,24 @@
                     <Grid x:Name="LayoutRoot" Background="Transparent">
                         <VisualStateManager.VisualStateGroups>
                             <VisualStateGroup x:Name="CommonStates">
-                                <VisualState x:Name="Active" />
-                                <VisualState x:Name="Inactive" />
+                                <VisualState x:Name="Active">
+                                    <Storyboard>
+                                        <DoubleAnimation
+                                            Storyboard.TargetName="IndeterminateAnimatedVisualPlayer"
```
Comment (ranjeshj) [inline dev/ProgressRing/ProgressRing.xaml:26]: >IndeterminateAnimatedVisualPlayer [](start = 67, length = 33)

Can we target the LayoutRoot's opacity ?

File: dev/ProgressRing/ProgressRing.xaml [hunk 2]
```diff
@@ -20,8 +20,24 @@
...
+                                            Storyboard.TargetProperty="Opacity"
+                                            To="1"
+                                            Duration="0:0:0.15" />
+                                    </Storyboard>
+                                </VisualState>
+                                <VisualState x:Name="Inactive">
+                                    <Storyboard>
```
Comment (ranjeshj) [inline dev/ProgressRing/ProgressRing.xaml:33]: I think just this should be enough. dont have to set it to 1 in Active. VSM should take care of that.

File: dev/ProgressBar/ProgressBar.idl [hunk 3]
```diff
@@ -27,6 +27,19 @@ unsealed runtimeclass ProgressBarTemplateSettings : Windows.UI.Xaml.DependencyOb
 
     Windows.UI.Xaml.Media.RectangleGeometry ClipRect;
     static Windows.UI.Xaml.DependencyProperty ClipRectProperty{ get; };
+
+    // TemplateSetting properties from WUXC for backwards compatibility.
```
Comment (ranjeshj) [inline dev/ProgressBar/ProgressBar.idl:31]: Please add tests for ProgressBar and ProgressRing where the template is set to the old template from WUXC. And another with the template for ProgressRing with a custom lottie animation source.

**Comment by ranjeshj** [comment]
>     static Windows.UI.Xaml.DependencyProperty TemplateSettingsProperty{ get; };

remove this and add the MUX_PROPERTY_NEEDS_DP_FIELD to the TemplateSettings property similar to other types.

---
Refers to: dev/ProgressBar/ProgressBar.idl:65 in a96399c. [](commit_id = a96399c729cbab1a2c07418298df0754303e586a, deletion_comment = False)

File: dev/ProgressRing/ProgressRing.xaml (hunk 2 above)
Comment (karenbtlai) [reply dev/ProgressRing/ProgressRing.xaml:33]: It is necessary for the FadeIn transition back to Opacity to 1. However, the debate can also be whether there should be a transition animation or not?

Comment (ranjeshj) [reply dev/ProgressRing/ProgressRing.xaml:33]: Ah i see.

---
In reply to: [406964416](https://github.com/microsoft/microsoft-ui-xaml/pull/2268#discussion_r406964416) [](ancestors = 406964416)

File: dev/ProgressRing/ProgressRing.xaml (hunk 1 above)
Comment (ranjeshj) [reply dev/ProgressRing/ProgressRing.xaml:26]: Actually this is probably better since with the determinate part, we might not want to do the same thing.

**Comment by ranjeshj** [comment]
@karenbtlai I separated out the fix for hiding on inactive into https://github.com/microsoft/microsoft-ui-xaml/pull/2272.

File: dev/Repeater/APITests/FlowLayoutTests.cs [hunk 4]
```diff
@@ -1601,22 +1601,29 @@ private void ValidateLayoutEstimations(ScrollOrientation scrollOrientation, Layo
...
+                            var minRowSpacing = om.ScrollOrientation == ScrollOrientation.Vertical ? lineSpacing : 0;
+                            var minColumnSpacing = om.ScrollOrientation == ScrollOrientation.Horizontal ? lineSpacing : 0;
+                            layout = new UniformGridLayout() { MinItemWidth = itemSize, MinItemHeight = itemSize, MinRowSpacing = minRowSpacing, MinColumnSpacing = minColumnSpacing };
+                        }
                         break;
 
                     case LayoutChoice.Flow:
-                        layout = new FlowLayoutDerived()
                         {
-                            MinColumnSpacing = lineSpacing,
```
Comment (ranjeshj) [inline dev/Repeater/APITests/FlowLayoutTests.cs:1612]: There seem to be a lot of unrelated changes in the PR. can you merge with master and force update your branch/PR ?

File: dev/ProgressBar/TestUI/ProgressBarReTemplatePage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            var layoutRoot = (Grid)VisualTreeHelper.GetChild(TestProgressBar, 0);
+
+            VisualStateManager.GetVisualStateGroups(layoutRoot)[0].CurrentStateChanged += this.ProgressBarPage_CurrentStateChanged;
+            VisualStateText.Text = VisualStateManager.GetVisualStateGroups(layoutRoot)[0].CurrentState.Name;
+
+            var progressBarRoot = VisualTreeHelper.GetChild(layoutRoot, 0);
+            var clip = VisualTreeHelper.GetChild(progressBarRoot, 0);
+            var stackPanel = VisualTreeHelper.GetChild(clip, 0);
+
+            //indicator.SizeChanged += this.Indicator_SizeChanged;
```
Comment (ranjeshj) [inline dev/ProgressBar/TestUI/ProgressBarReTemplatePage.xaml.cs:29]: nit: Remove
